rules-base/api/src/main/java/org/jboss/windup/rules/files/condition/File.java

```diff
@@ -0,0 +1,264 @@
+        return f;
+    }
+
+    /**
+     * Optionally specify the variable name to use for the output of this condition
+     */
+    public ConditionBuilder as(String variable)
+    {
+        Assert.notNull(variable, "Variable name must not be null.");
+        this.setOutputVariablesName(variable);
+        return this;
+    }
+
+    @Override
+    public Set<String> getRequiredParameterNames()
+    {
+        Set<String> result = new HashSet<>();
+        if (filenamePattern != null)
+            result.addAll(filenamePattern.getRequiredParameterNames());
+        return result;
+    }
+
+    @Override
+    public void setParameterStore(ParameterStore store)
+    {
+        if (filenamePattern != null)
+            filenamePattern.setParameterStore(store);
+    }
+
+    @Override
+    protected String getVarname()
+    {
+        return getOutputVariablesName();
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    protected boolean evaluateAndPopulateValueStores(GraphRewrite event, EvaluationContext context, final FrameCreationContext frameCreationContext)
+    {
+        return evaluate(event, context, new EvaluationStrategy()
+        {
+            private LinkedHashMap<String, List<WindupVertexFrame>> variables;
+
+            @Override
+            @SuppressWarnings("rawtypes")
+            public void modelMatched()
+            {
+                this.variables = new LinkedHashMap<String, List<WindupVertexFrame>>();
+                frameCreationContext.beginNew((Map) variables);
+            }
+
+            @Override
+            public void modelSubmitted(WindupVertexFrame model)
+            {
+                Maps.addListValue(this.variables, getVarname(), model);
+            }
+
+            @Override
+            public void modelSubmissionRejected()
+            {
+                frameCreationContext.rollback();
+            }
+        });
+    }
+
+    @Override
+    protected boolean evaluateWithValueStore(GraphRewrite event, EvaluationContext context, final FrameContext frameContext)
+    {
+        boolean result = evaluate(event, context, new NoopEvaluationStrategy());
+
+        if (result == false)
+            frameContext.reject();
+
+        return result;
+    }
+
+    private boolean evaluate(final GraphRewrite event, final EvaluationContext context,
+                final EvaluationStrategy evaluationStrategy)
+    {
+        final ParameterStore store = DefaultParameterStore.getInstance(context);
+        final GraphService<FileReferenceModel> fileReferenceService = new GraphService<>(event.getGraphContext(), FileReferenceModel.class);
+
+        // initialize the input
+        List<FileModel> fileModels = new ArrayList<>();
+        fromInput(fileModels, event);
+        fileNameInput(fileModels, event, store);
+        allInput(fileModels, event, store);
+
+        final List<FileReferenceModel> results = new ArrayList<>();
```

Comment:
I was not sure if File condition should return FileModel or FileReferenceModel. I ended up with FileReferenceModel because in the future it may be possible to provide additional information about this match and extend the FileReferenceModel.